near-contract-standards/src/non_fungible_token/approval/approval_impl.rs

```diff
@@ -64,19 +62,17 @@ impl NonFungibleTokenApproval for NonFungibleToken {
         refund_deposit(storage_used);
 
         // if given `msg`, schedule call to `nft_on_approve` and return it. Else, return None.
-        if let Some(msg) = msg {
-            Some(ext_approval_receiver::nft_on_approve(
+        msg.map(|msg| {
```

Comment:
This is still so confusing to me. :D

Reply:
Instead of manually constructing both the `Some` and `None` variants, it just transforms the inner value